### ISSUE TITLE
reversed_queries_test.py::test_queries_during_upgrade mis-populates tableted keyspace

### ISSUE THREAD

**Issue by xemul** [scylladb/scylladb#16745]
build id 7ea78454c2abfa05e04884e24c276a41865e6dd1
release 0.20240109.f4f724921cc7
https://jenkins.scylladb.com/job/scylla-master/job/byo/job/dtest-byo/93

```
INFO  2024-01-10 11:31:43,699 [shard 0:main] init - loading tablet metadata
INFO  2024-01-10 11:31:43,700 [shard 0:main] init - loading non-system sstables
ERROR 2024-01-10 11:31:43,700 [shard 0:main] database - Skipping: test_reversed_queries. Exception occurred when loading system table keyspaces: exceptions::configuration_exception (initial_tablets is reserved name for NetworkTopologyStrategy)
ERROR 2024-01-10 11:31:43,703 [shard 0:main] database - Skipping: test_reversed_queries. Exception occurred when loading system table tables: data_dictionary::no_such_keyspace (Can't find a keyspace test_reversed_queries)
ERROR 2024-01-10 11:31:43,706 [shard 1:main] database - Skipping: test_reversed_queries. Exception occurred when loading system table keyspaces: exceptions::configuration_exception (initial_tablets is reserved name for NetworkTopologyStrategy)
ERROR 2024-01-10 11:31:43,708 [shard 1:main] database - Skipping: test_reversed_queries. Exception occurred when loading system table tables: data_dictionary::no_such_keyspace (Can't find a keyspace test_reversed_queries)
```

which is weird. Probably will go away with #16364

**Comment by xemul** [comment]
Here's what happens.

First, on older scylla test does CREATE KEYSPACE ... REPLICATION = { initial_tablets: 10 } and this very "initial_tablets: 10" gets into system schema into replication line, because older scylla treats "initial_tablets" as datacenter name (see, e.g. [this](https://github.com/scylladb/scylladb/pull/16364#discussion_r1422655983) discussion). Second, newer version of scylla starts over pre-existing schema and loads the "initial_tablets: 10" from it, thus stepping on the reserved name check

**Comment by xemul** [comment]
So this particular bug will be transparently fixed by #16364, but we'll probably step on something else
